### PR TITLE
initial upload

### PR DESCRIPTION
**Pull request by vincerubinetti**

### CODE REVIEW

File: 404.html
Comment (d33bs) [inline 404.html:1]: Would it be possible to call this file `301.html` (for moved permanently) instead of `404.html` (not found)? Or would that break functionality entirely? If it must be `404.html`, does this mean a 404 is initially encountered when following links from this repo (followed by a 301 or 200)?

File: redirect.js
Comment (d33bs) [inline redirect.js:22]: Does this have a potential to be blocked by a client due to built-in security checks? Would [`location.assign()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/location#example_1_navigate_to_a_new_page) offer any benefits?

**Review by d33bs — COMMENTED** [review]
Thank you for including me on this and the related redirects repo PR @vincerubinetti ! I left a couple comments about the content (some of which may have been more appropriate for the redirects repo, apologies).

File: 404.html
Comment (vincerubinetti) [reply 404.html:1]: This is the only way to make it work, as far as I know, with GitHub Pages. They don't support HTTP-level redirects like that.

File: redirect.js
Comment (vincerubinetti) [reply redirect.js:22]: This is the typical way to navigate for as long as I can remember. I can't find any evidence that there's a difference between location.assign/replace and location.href = "". Did you see something? I also have no experience with this being blocked by extensions/flags/errors, or heard of anyone else having that issue.

Comment (d33bs) [reply redirect.js:22]: I'm less familiar with how this might operate within various clients so would defer to your best judgement here.

**Review by d33bs — APPROVED** [review]

**Pull request merged**
